packages/discovery-provider/src/challenges/play_count_milestones_challenge.py

```diff
@@ -0,0 +1,215 @@
+            step_count = challenge.current_step_count
+            for milestone in sorted(PLAY_MILESTONES.keys()):
+                if step_count is not None and step_count >= milestone:
+                    if milestone not in completed_milestones:
+                        completed_milestones.append(milestone)
+
+        return completed_milestones
+
+    def update_user_challenges(
+        self,
+        session: Session,
+        event: str,
+        user_challenges: List[UserChallenge],
+        step_count: Optional[int],
+        event_metadatas: List[FullEventMetadata],
+        starting_block: Optional[int],
+    ):
+        """
+        Update user challenges based on their play count milestones.
+        This method is called by the ChallengeManager when processing events.
+        """
+        if event != ChallengeEvent.track_played:
+            return
+
+        for user_challenge in user_challenges:
+            user_id = user_challenge.user_id
+            play_count = self.get_user_play_count_2025(session, user_id)
+            completed_milestones = self._get_completed_milestones(session, user_id)
+            sorted_milestones = sorted(PLAY_MILESTONES.keys())
+            milestone = None
+
+            if not completed_milestones:
+                milestone = sorted_milestones[0]
+            else:
+                # Find the next milestone after the highest completed one
+                highest_completed = max(completed_milestones)
+                next_index = sorted_milestones.index(highest_completed) + 1
+                if next_index < len(sorted_milestones):
+                    milestone = sorted_milestones[next_index]
+                else:
+                    # This is a challenge for a milestone beyond the final one
+                    # This shouldn't happen, but if it does, use the final milestone
+                    milestone = FINAL_MILESTONE
+
+            if milestone is None:
+                continue
+
+            user_challenge.current_step_count = play_count
+
+            if play_count >= milestone:
+                user_challenge.amount = int(PLAY_MILESTONES[milestone])
+                user_challenge.is_complete = True
+
+    def generate_specifier(self, session: Session, user_id: int, extra: Dict) -> str:
+        """
+        Generate a unique specifier for the user's challenge.
+
+        Format: hex_user_id_timestamp
+        Example: a1b2c3_20250315123045
+
+        Simple temporal specifier without embedding milestone logic.
+        """
+        completed_milestones = self._get_completed_milestones(session, user_id)
+
+        # If user has already completed the final milestone, don't create a new challenge
+        if FINAL_MILESTONE in completed_milestones:
+            return f"{hex(user_id)[2:]}_dummy_{datetime.now().strftime('%Y%m%d%H%M%S')}"
+
+        # For simplicity, just use timestamp for uniqueness
+        timestamp = datetime.now().strftime("%Y%m%d%H%M%S")
```

Comment:
might want to confirm that this will fit inside 21 bytes (for rewards manager PDA)